src/components/Modal/index.android.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import getModalStyles from '../../styles/getModalStyles';
+import CONST from '../../CONST';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import ModalSkeleton from './ModalSkeleton';
+
+const propTypes = {
+    // Callback method fired when the user requests to close the modal
+    onClose: PropTypes.func.isRequired,
+
+    // State that determines whether to display the modal or not
+    isVisible: PropTypes.bool.isRequired,
+
+    // Modal contents
+    children: PropTypes.node.isRequired,
+
+    // Callback method fired when the user requests to submit the modal content.
+    onSubmit: PropTypes.func,
+
+    // Callback method fired when the modal is hidden
+    onModalHide: PropTypes.func,
+
+    // Style of modal to display
+    type: PropTypes.oneOf([
+        CONST.MODAL.MODAL_TYPE.CENTERED,
+        CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED,
+        CONST.MODAL.MODAL_TYPE.POPOVER,
+        CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED,
+    ]),
+
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    onSubmit: null,
+    type: '',
+    onModalHide: () => { },
+};
+
+const Modal = (props) => {
+    const {
+        useNativeDriver,
+        modalStyle,
+        modalContainerStyle,
+        swipeDirection,
+        animationIn,
+        animationOut,
+        shouldAddTopSafeAreaPadding,
+        shouldAddBottomSafeAreaPadding,
+        hideBackdrop,
+    } = getModalStyles(props.type, {
+        windowWidth: props.windowWidth,
+        windowHeight: props.windowHeight,
+        isSmallScreenWidth: props.isSmallScreenWidth,
+    });
+    return (
+        <ModalSkeleton
```

Comment:
This can just be called `Modal` no need to call it a `Skeleton`

Reply:
Then, there would be two components with the same name

Reply:
maybe name it `BaseModal` similar to what we do here?
<img width="211" alt="Screen Shot 2021-02-06 at 9 36 21 AM" src="https://user-images.githubusercontent.com/32969087/107128060-d5e98e80-685e-11eb-80de-cd744c96231a.png">